lib/spack/spack/modules/common.py

```diff
@@ -734,11 +734,13 @@ def environment_modifications(self):
             _check_tokens_are_valid(x.name, message=msg)
             # Transform them
             x.name = spec.format(x.name, transform=transform)
-            try:
-                # Not every command has a value
-                x.value = spec.format(x.value)
-            except AttributeError:
-                pass
+            # Skip value format if set raw
+            if not isinstance(x, (spack.util.environment.SetEnv)) or not x.raw:
```

Comment:
Can we extract the condition in a method, to express intent? Something like:
```
if self.modification_needs_formatting(x):
    ...
```

Reply:
I have added the suggested `modification_needs_formatting` function. Hope this is done as expected. Let me know if I have things to refine.